config.go

```diff
@@ -28,19 +32,29 @@ type RefreshPeriod struct {
 }
 
 type Config struct {
-	CacheDir        string          `yaml:"cache_dir"`
-	Port            int             `yaml:"port"`
-	Repos           map[string]Repo `yaml:"repos,omitempty"`
-	PurgeFilesAfter int             `yaml:"purge_files_after"`
-	DownloadTimeout int             `yaml:"download_timeout"`
-	Prefetch        *RefreshPeriod  `yaml:"prefetch"`
-	HttpProxy       string          `yaml:"http_proxy"`
-	UserAgent       string          `yaml:"user_agent"`
-	LogTimestamp    bool            `yaml:"set_timestamp_to_logs"`
+	CacheDir        string           `yaml:"cache_dir"`
+	Port            int              `yaml:"port"`
+	Repos           map[string]*Repo `yaml:"repos,omitempty"`
+	PurgeFilesAfter int              `yaml:"purge_files_after"`
+	DownloadTimeout int              `yaml:"download_timeout"`
+	Prefetch        *RefreshPeriod   `yaml:"prefetch"`
+	HttpProxy       string           `yaml:"http_proxy"`
+	UserAgent       string           `yaml:"user_agent"`
+	LogTimestamp    bool             `yaml:"set_timestamp_to_logs"`
 }
 
 var config *Config
 
+func initURLsChannel(name string, repo *Repo) {
+	in := make(chan chan []string)
+	go func() {
+		for out := range in {
```

Comment:
Is it needed to be stopped at some point? Or it is ok to keep it running forever?

Reply:
I think the only time you'd want to stop it is if you won't be handling any more URLs requests, so on exit in other words. It blocks until it gets a message over the channel, so it's not busy looping in the background or anything.